src/main/java/org/commcare/formplayer/services/MediaMetaDataService.java

```diff
@@ -20,6 +20,14 @@ public int purge() {
         return mediaMetaDataRepo.deleteMetaDataWithoutFormSessionId();
     }
 
+    public MediaMetadataRecord findByFormSessionId(String sessionId) {
+        return mediaMetaDataRepo.findByFormSessionId(sessionId).get();
+    }
+
+    public MediaMetadataRecord findByFilePath(String filePath) {
```

Comment:
this method looks unused at the moment.

Reply:
Using this now in testing that metadata is updated when file is reuploaded.

Reply:
As a best practice we should restrain from adding public methods in main code which are only used in tests. I think we can avoid both `findByFormSessionId` and  `findByFilePath` here by directly querying using `fileID`. You should be able to derive the `fileId` from the path here by getting the file name from the file path and then removing the extension.